lib/Query/Resolver.php

```diff
@@ -45,48 +45,56 @@ public function noQuote($noQuote = true)
-     * Migrate table structure changes to database
-     *
-     * @return bool
-     * @throws \Doctrine\DBAL\Schema\SchemaException
-     * @throws \Spot\Exception
-     */
-    public function migrate()
-    {
-        // Mapper knows currently set entity
-        $entity = $this->mapper->entity();
-        $table = $entity::table();
-        $fields = $this->mapper->entityManager()->fields();
-        $fieldIndexes = $this->mapper->entityManager()->fieldKeys();
-        $connection = $this->mapper->connection();
-
-        $schemaManager = $this->mapper->connection()->getSchemaManager();
-        $tableObject = $schemaManager->listTableDetails($table);
-        $tableObjects[] = $tableObject;
-        $schema = new \Doctrine\DBAL\Schema\Schema($tableObjects);
-
-        $tableColumns = $tableObject->getColumns();
-        $tableExists = !empty($tableColumns);
-        if ($tableExists) {
-            // Update existing table
-            $existingTable = $schema->getTable($table);
-            $newSchema = $this->migrateCreateSchema();
-            $queries = $schema->getMigrateToSql($newSchema, $connection->getDatabasePlatform());
-        } else {
-            // Create new table
-            $newSchema = $this->migrateCreateSchema();
-            $queries = $newSchema->toSql($connection->getDatabasePlatform());
-        }
-
-        // Execute resulting queries
-        $lastResult = false;
-        foreach ($queries as $sql) {
-            $lastResult = $connection->exec($sql);
-        }
-
-        return $lastResult;
-    }
+	/**
+	 * Migrate table structure changes to database
+	 *
+	 * @return bool
+	 * @throws \Doctrine\DBAL\Schema\SchemaException
+	 * @throws \Spot\Exception
+	 */
+	public function migrate()
+	{
+		// Mapper knows currently set entity
+		$entity = $this->mapper->entity();
+		$table = $entity::table();
+		$tableCreated = false;
+		$fields = $this->mapper->entityManager()->fields();
+		$fieldIndexes = $this->mapper->entityManager()->fieldKeys();
+		$connection = $this->mapper->connection();
+
+		$schemaManager = $this->mapper->connection()->getSchemaManager();
+		$tableObject = $schemaManager->listTableDetails($table);
+		$tableObjects[] = $tableObject;
+		$schema = new \Doctrine\DBAL\Schema\Schema($tableObjects);
+
+		$tableColumns = $tableObject->getColumns();
+		$tableExists = !empty($tableColumns);
+		if ($tableExists) {
+			// Update existing table
+			$existingTable = $schema->getTable($table);
+			$newSchema = $this->migrateCreateSchema();
+			$queries = $schema->getMigrateToSql($newSchema, $connection->getDatabasePlatform());
+		} else {
+			// Create new table
+			$newSchema = $this->migrateCreateSchema();
+			$tableCreated = true;
+			$queries = $newSchema->toSql($connection->getDatabasePlatform());
+		}
+
+		// Execute resulting queries
+		$lastResult = false;
+		foreach ($queries as $sql) {
+			$lastResult = $connection->exec($sql);
+		}
+
+		//
+		if ($tableCreated)
+		{
+			$entity::bootstrap();
```

Comment:
It's a hook that can be used to populate the database with default data (primarily) on creation. For example it could be a users table adding default credentials, or a countries table adding in a default dataset. This way you can let the migrator create and auto-populate any databases that need it.